scripts/testing/README.md

```diff
@@ -0,0 +1,112 @@
+## Automated testing
+
+### Installation
+
+Atomated testing CLI depends on `gcloud` and `vault` CLIs so you need to install them. 
+
+#### Google cloud CLI
+Follow the [offical installation](https://cloud.google.com/sdk/docs/install) guide. Once installed, run `gcloud auth login` and authorize using SSO. 
+
+#### Vault
+Download and install the Hashicorp Vault client version 1.14.1 or newer or via brew with `brew install vault`
+1. `brew install vault`.
+2. [Generate](https://github.com/elastic/infra/tree/master/docs/vault#github-auth) a github token to authorize in Vault.
+3. Run `vault login --method github` to authorize your Vault cli.
+
+#### Connector service
+Run `make clean install` to generate executable files in `./bin` folder
+
+### Usage
+
+***Note:*** The working directory for `./bin/test-connectors` is `scripts/testing` so when you need to point to a file you should use it as a root directory. 
+e.g. `./bin/test-connectors run-test my-testing-environment-name --es-version 8.12-SNAPSHOT --test-case scenarios/clients/spo_full_and_incremental_syncs.yml`
+
+Run `./bin/test-connectors --help` or `./bin/test-connectors {command name} --help` to get more information about the cli. 
+
+#### Running test with Elastic cloud deployment
+If you want to run your test suite using a cloud Elasticsearch deployment follow the next steps: 
+1. Create a cloud deployment
+2. Download a credentials file (or create a new user)
+3. Run `./bin/test-connectors run-test my-testing-environment-name --es-host {host} --es-username {user name} --es-password {password} --test-case {path to the test case file}`
+
+#### Running test with local Elasticsearch
+If you want to run your tests with local Elasticsearch you need to specify `--es-version` option. Like
+
+`./bin/test-connectors run-test my-testing-environment-name --es-version 8.12-SNAPSHOT --test-case {path to the test case file}`
+
+In this case, the cli will deploy an Elasticsearch instance in the same VM where the connector service will be running. 
+
+#### Running test with a specific Connector service version
+
+You can use any git reference such as commit sha, a tag, or a branch name. The cli will pull the defined git reference and run `make clean install`. 
+
+Example: `./bin/test-connectors run-test my-testing-environment-name --es-version 8.12-SNAPSHOT --connectors-ref 8.12 --test-case {path to the test case file}`
+
+#### Keeping your VM running when the tests passed
+Sometimes it's useful to get access to the logs or make some changes in the code and run tests again. The CLI will print a list of useful commands you can use to access the VM resources like: 
+
+`Access logs: gcloud compute ssh {my-testing-environment-name} --zone {VM zone} --command "tail -f ~/service.log"`
+
+To automatically delete the VM you need to use `--delete` option. 
```

Comment:
Curious - do you think it should delete by default instead, so that it's harder to miss that part that you need to delete stuff manually?

Reply:
I assumed that when run a test and it fails you will want to SSH to the VM or at least look at the logs. All of this will be impossible if VM is gone.  One of the next steps is to have scheduled pipelines in Buildkite and one of them will be responsible for cleaning up the old VMs.